### PR TITLE
Refactored inccounter

### PR DESCRIPTION
**Pull request by melatron**

### CODE REVIEW

File: contracts/native/inccounter/impl.go [hunk 1]
```diff
@@ -53,7 +56,8 @@ func init() {
 }
 
 func initialize(ctx coretypes.Sandbox) (dict.Dict, error) {
-	ctx.Log().Debugf("inccounter.init in %s", ctx.ContractID().Hname().String())
+	ctx.Contract().String()
```
Comment (lunfardo314) [inline contracts/native/inccounter/impl.go]: ?

File: contracts/native/inccounter/impl.go [hunk 2]
```diff
@@ -125,11 +133,15 @@ func incCounterAndRepeatMany(ctx coretypes.Sandbox) (dict.Dict, error) {
...
+	iota1 := ledgerstate.NewColoredBalances(map[ledgerstate.Color]uint64{ledgerstate.ColorIOTA: 1})
+	meta := &coretypes.SendMetadata{
+		TargetContract: ctx.Contract(),
+		EntryPoint:     coretypes.Hn(FuncIncAndRepeatMany),
+	}
+	opts := coretypes.SendOptions{TimeLock: 1 * 60}
+	succ := ctx.Send(ctx.ChainID().AsAddress(), iota1, meta, opts)
+
+	if succ {
 		ctx.Log().Debugf("PostRequestToSelfWithDelay. remaining repeats = %d", numRepeats-1)
```
Comment (lunfardo314) [inline contracts/native/inccounter/impl.go]: the message must be changed accordingly

File: contracts/native/inccounter/inccounter_test.go [hunk 3]
```diff
@@ -85,7 +86,7 @@ func TestIncWith1Post(t *testing.T) {
 	checkCounter(chain, 17)
 
 	req := solo.NewCallParams(incName, FuncIncAndRepeatOnceAfter5s).
-		WithTransfer(balance.ColorIOTA, 1)
+		WithTransfer(ledgerstate.ColorIOTA, 1)
```
Comment (lunfardo314) [inline contracts/native/inccounter/inccounter_test.go]: WithIotas(1)

**Review by lunfardo314 — CHANGES_REQUESTED** [review]

**Comment by BugFreeSoftware** [comment]
Most of these changes were already merged as part of my refactoring of the entire contracts folder

**Pull request closed without merging**